wrappers/node/demo/faber.js

```diff
@@ -1,246 +1,174 @@
-  assert(proofState)
-  assert(proof)
-  logger.info(`proofState = ${JSON.stringify(proofProtocolState)}`)
-  logger.info(`vcxProof = ${JSON.stringify(vcxProof)}`)
+    await vcxClient.credentialIssue(schemaAttrs, 'DemoCredential123', connectionName, options.revocation)
+
+    const proofAttributes = [
+      {
+        names: ['name', 'last_name', 'sex'],
+        restrictions: [{ issuer_did: issuerDid }]
+      },
+      {
+        name: 'date',
+        restrictions: { issuer_did: issuerDid }
+      },
+      {
+        name: 'degree',
+        restrictions: { 'attr::degree::value': 'maths' }
+      },
+      {
+        name: 'nickname',
+        self_attest_allowed: true
+      }
+    ]
+
+    const proofPredicates = [
+      { name: 'age', p_type: '>=', p_value: 20, restrictions: [{ issuer_did: agentProvision.institution_did }] }
+    ]
+
+    logger.info('#19 Create a Proof object')
+    const vcxProof = await Proof.create({
+      sourceId: '213',
+      attrs: proofAttributes,
+      preds: proofPredicates,
+      name: 'proofForAlice',
+      revocationInterval: { to: Date.now() }
+    })
+
+    logger.info('#20 Request proof of degree from alice')
+    await vcxProof.requestProof(connectionToAlice)
+
+    logger.info('#21 Poll agency and wait for alice to provide proof')
+    let proofProtocolState = await vcxProof.getState()
+    logger.info(`vcxProof = ${JSON.stringify(vcxProof)}`)
+    logger.info(`proofState = ${proofProtocolState}`)
+    while (proofProtocolState !== StateType.Accepted) {
+      // even if revoked credential was used, vcxProof.getState() should in final state return StateType.Accepted
+      await sleepPromise(2000)
+      await vcxProof.updateState()
+      proofProtocolState = await vcxProof.getState()
+      logger.info(`proofState=${proofProtocolState}`)
+    }
 
-  logger.info('#28 Check if proof is valid.')
-  if (proofState === ProofState.Verified) {
-    logger.warn('Proof is verified.')
-    assert(options.revocation === false)
-  } else if (proofState === ProofState.Invalid) {
-    logger.warn('Proof verification failed, credential has been revoked.')
-    assert(options.revocation === true)
-  } else {
-    logger.error(`Unexpected proof state '${proofState}'.`)
-    process.exit(-1)
+    logger.info('#27 Process the proof provided by alice.')
+    const { proofState, proof } = await vcxProof.getProof(connectionToAlice)
+    assert(proofState)
+    assert(proof)
+    logger.info(`proofState = ${JSON.stringify(proofProtocolState)}`)
+    logger.info(`vcxProof = ${JSON.stringify(vcxProof)}`)
+
+    logger.info('#28 Check if proof is valid.')
+    if (proofState === ProofState.Verified) {
+      logger.warn('Proof is verified.')
+      assert(options.revocation === false)
+    } else if (proofState === ProofState.Invalid) {
+      logger.warn('Proof verification failed, credential has been revoked.')
+      assert(options.revocation === true)
```

Comment:
I guess a proof in this state can be received for another reason, but not in any scenario within this demo (unless we use it to test compatibility with other agents)...

Reply:
Good point. I've changed the message a bit.